in_toto/runlib.py

```diff
@@ -903,37 +930,48 @@ def in_toto_record_stop(step_name, product_list, signing_key=None,
     keyid = link_metadata.signatures[0]["keyid"]
     gpg_pubkey = securesystemslib.gpg.functions.export_pubkey(
         keyid, gpg_home)
-    verification_key = gpg_pubkey
+    verification_key = GPGKey.from_dict(gpg_pubkey)
+
+  link_metadata.verify_sigs([verification_key], 1)
 
-  link_metadata.verify_signature(verification_key)
+  LOG.info("Extracting Link from metadata...")
+  link = link_metadata.get_payload()
+  use_metablock = isinstance(link_metadata, Metablock)
 
   # Record products if a product path list was passed
   if product_list:
     LOG.info("Recording products '{}'...".format(", ".join(product_list)))
 
-  link_metadata.signed.products = record_artifacts_as_dict(
+  link.products = record_artifacts_as_dict(
       product_list, exclude_patterns=exclude_patterns, base_path=base_path,
       follow_symlink_dirs=True, normalize_line_endings=normalize_line_endings,
       lstrip_paths=lstrip_paths)
 
-  link_metadata.signatures = []
+  if use_metablock:
+    LOG.info("Generating link metadata using Metablock...")
+    link_metadata = Metablock(signed=link)
+  else:
```

Comment:
`AnyMetadataDeserializer` already raises a [InvalidMetadata exception](https://github.com/in-toto/in-toto/pull/503/files#diff-d036bafdd463a770d30e4159e24f9f2c5dce02546cc59cc4f85dfeba1bb12aa8R64) while loading. 

There is signature verification and payload extraction step in between loading of link metadata and generation of new metadata.
So it would be better if I have to check for both metadata after the [`AnyMetadata.from_file()`](https://github.com/in-toto/in-toto/pull/503/files#diff-40f0ea7898882c994402200f6f9c27e7b8e1934971eda4ba81b853b3d9673466R905) command.